docs/source/contributing.rst

```diff
@@ -152,6 +152,7 @@ The json description of a generic processor must contain the attributes:
   - ``arguments``: An object describing arguments passed to the tag.
   - ``template_parameters``: An object describing template parameters.
   - (Optional) ``template_name``: A custom name for the html-template to use. Defaults to the processor name otherwise.
+  - (Optional) ``tag_argument``: The text given at the beginning of a tag (e.g. the tag argument for ``{image file-path="example.png"}`` is ``image``). This is only necessary for processors with different names sharing the same resources (e.g. both ``image-container`` and ``image-tag`` share the ``image.html`` template).
```

Comment:
Confused why this talks about tag label then goes on about HTML templates?

Reply:
An image how has multiple processors introduced in this PR, `image-container` and `image-tag`, but both of these processors are used in the markdown files like so:
```
{image file-path="an-image.png"}
```
Therefore I introduced a new optional attribute to the processor definitions in the JSON file to specify the opening tag argument for a processor, the html part is simply giving an example of why you would do this.

Reply:
Is `tag_argument` tied to the HTML template that is used? I'm confused to why it doesn't say:

> (e.g. both ``image-container`` and ``image-tag`` share the ``tag_argument``).

Reply:
OH I see your point, will change to your suggestion for clarification :+1: